### DISCUSSION TITLE
Doctrine + Postgres duplicate records on insert

### DISCUSSION THREAD

**Discussion by JPradaa** [symfony/symfony#52299] [Q&A]
Hello everyone

I'm starting a project with Symfony 6.3 and Postgres 14
When I create a new entity and save it in the DB using "persist" and "flush" 2 identical records are created in the DB

**Controller**
```
#[Route('/{client_code}', name: 'app_client')]
    public function clientCode(Request $request, EntityManagerInterface $em): Response
    {

        $new_client = new Client();
        $new_client->setName('Test01');
        $new_client->setCode('TEST01');

        $em->persist($new_client);
        $em->flush();        
        
        return $this->render('client/index.html.twig', [
            'controller_name' => 'ClientController',
        ]);
    }
```
    
**Entity**
```
#[ORM\Table(name: 'live.client')]
#[ORM\Entity(repositoryClass: ClientRepository::class)]
class Client
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(options: ["autoincrement" => true])]
    private ?int $id = null;

    #[ORM\Column(length: 512)]
    private ?string $name = null;

    #[ORM\Column(length: 512)]
    private ?string $code = null;
    
   public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getCode(): ?string
    {
        return $this->code;
    }

    public function setCode(string $code): static
    {
        $this->code = $code;

        return $this;
    }
}
```

Using MySQL the same code inserts a single record, but with Postgres 2 records are inserted

The biggest difference I see is that Doctrine, in the case of Postgres, performs a
`SELECT NEXTVAL('live.client_id_seq')`
to get the next value in the sequence and generate an INSERT using that value, instead of leaving that task to the DB 
`INSERT INTO live.client (id, name, code) VALUES (25, 'Cliente name', 'CLIENT_CODE')`

I have tried it on Windows and Linux, using different versions of Symfony 6.x, ... and in all cases I get the same result

If once the page is loaded (and 2 records created in the CLIENT table) I go to the profiler and later return to the page using the browser's "back" button, then only 1 record is inserted instead of 2

Does anyone know what could be happening and how to fix the problem?

Thank you very much for your help
All the best

**Comment by ghost** [comment]
Please ask this question in doctrine/orm repository.

**Reply by JPradaa** [reply]
Hi @javaDeveloperKid 

But I'm not sure if the problem is related to Symfony or Doctrine...

Kind regards

**Reply by ghost** [reply]
Described problem looks like impossible to happen. You have to debug deeper because I doubt anyone can help with given details

**Comment by ghost** [comment]
Hi,

For me not double duplicate. Postgresql 15.4 ;)
